### PR TITLE
WIP: iOS Component Documentation

### PR DESCRIPTION
**Pull request by robbiet480**
Todo:
- [x] component docs
- [x] device_tracker docs
- [x] How location works
- [x] Notifications
  - [x] Basic usage
  - [x] Advanced usage
      - [x] Subtitle
      - [x] Badge
      - [x] Sound
      - [x] Attachments
      - [x] Actions
      - [x] Content Extension (map, camera)
  - [x] Architecture
  - [x] Rate Limits

### CODE REVIEW

**Comment by robbiet480** [comment]
@balloob @fabaff I think i'm going to need an entire section dedicated to this, like Getting Started. There's a lot to cover. For example, I think that notifications should be broken up into multiple sections. I'm basically writing a manual for the entire app. Let me know how best to proceed.

Also, this is not all of the docs. Maybe 50%. I stopped because I realized that while the content is great the layout is not working. I'm like 6 header levels deep in a few sections of the notify docs.

**Comment by robbiet480** [comment]
Todo:
- [ ] device_tracker docs
- [ ] How location works
- [ ] sensor docs
- [ ] Notifications
  - [ ] Content Extension (map, camera)

**Comment by robbiet480** [comment]
@fabaff @balloob I’ll be afk most of the evening but will have my phone. Can you do me a favor and review the iOS documentation and merge it if you think it’s good to go. I just want to get something out there for the moment and will add to it/fill in gaps over time.

File: source/_ecosystem/ios/notifications.markdown [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+sidebar: true
+comments: false
+sharing: true
+footer: true
+---
+
+The `ios` notify platform enables sending push notifications to the Home Assistant iOS app.
+
+## {% linkable_title Setup %}
+```yaml
```
Comment (balloob) [inline source/_ecosystem/ios/notifications.markdown]: Add a new line before

File: source/_ecosystem/ios/notifications.markdown [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# Example configuration.yaml entry
+notify:
+  - platform: ios
+```
+
+Configuration variables:
+
+- **name** (*Optional*): The name of the service. Not recommended to add this.
```
Comment (balloob) [inline source/_ecosystem/ios/notifications.markdown]: Why not recommended?

File: source/_ecosystem/ios/notifications/actions.markdown [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+Actionable notifications allow you to attach 1-4 custom buttons to a notification. When one of the actions is selected Home Assistant will be notified which action was chosen. This allows you to build complex automations.
+
+Examples of actionable notifications:
+
+- A notification is sent whenever motion is detected in your home while you are away or asleep. You can add an action to Sound Alarm. When tapped, Home Assistant is notified that the `sound_alarm` action was selected. You can add an automation to sound the burglar alarm whenever this event is seen.
+- Someone rings your front door bell. You can send an action to lock or unlock your front door. When tapped, a notification is sent back to Home Assistant upon which you can build automations.
+- Send a notification whenever your garage door opens with actions to open and close the garage.
+
+![Actions example](/images/ios/actions.png)
```
Comment (balloob) [inline source/_ecosystem/ios/notifications/actions.markdown]: Prefer if you use our img format:

```html
<p class='img'>
  <img src='/images/ios/actions.png' />
  Actionable notifications allow the user to send a command back to Home Assistant.
</p>
```

Also, please crop away all the not-needed part of the image, it takes a lot of height.

File: source/_ecosystem/ios/notifications/architecture.markdown [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+title: "Architecture"
+description: "The push notification system layout"
+date: 2016-10-25 15:00:00 -0700
+sidebar: true
+comments: false
+sharing: true
+footer: true
+---
+
+![Architecture](/images/ios/PushNotificationLayout.png)
```
Comment (balloob) [inline source/_ecosystem/ios/notifications/architecture.markdown]: Use our img wrapper

File: source/_ecosystem/ios/notifications/attachments.markdown [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+```
+
+Notes:
+* The thumbnail of the notification will be the media at the `url`.
+* The notification content is the media at the `url`.
+* Attachment can be used with custom push notification categories.
+
+## Example
+
+![Attachment](/images/ios/attachment.png)
```
Comment (balloob) [inline source/_ecosystem/ios/notifications/attachments.markdown]: Clip those images…

File: source/_ecosystem/ios/notifications/basic.markdown [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+title: "Basic Notifications"
+description: "Basic notes about iOS notifications"
+date: 2016-10-25 15:00:00 -0700
+sidebar: true
+comments: false
+sharing: true
+footer: true
+---
+
+The iOS notify platform accepts the standard `title`, `message` and `target` parameters.
```
Comment (balloob) [inline source/_ecosystem/ios/notifications/basic.markdown]: No screenshots on this page? This is like one of the first notifications pages people go too. It's fine to re-use an image

File: source/_ecosystem/ios/notifications/content_extensions.markdown [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+title: "Dynamic content"
+description: "Extend your notifications with dynamic content"
+date: 2016-10-25 15:00:00 -0700
+sidebar: true
+comments: false
+sharing: true
+footer: true
+---
+
+With the new Content Extension feature found in iOS 10, dynamic content can now be displayed as part of a notification without opening an app.
```
Comment (balloob) [inline source/_ecosystem/ios/notifications/content_extensions.markdown]: As a developer this is different, however, will it be different from attachments for a user ?

File: source/_ecosystem/ios/notifications/privacy_security_rate_limits.markdown [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+## {% linkable_title Privacy %}
+
+No notification content is stored on remote servers. Only the required push registration data and a simple counter of the total number of push notifications sent per day per device (for rate limiting purposes) is kept.
+
+## {% linkable_title Rate limiting %}
+
+Currently, you are allowed to send a maximum of 150 push notifications per day per device. This is to ensure that the service remains cheap to maintain. In the future we may add support for upgrading to allow more notifications. The rate limit resets at midnight UTC daily. When a notification is sent your current rate limits (including sent notifications and notifications remaining for the day) will be output to your Home Assistant logs. If an error occurs while sending a notification your rate limit will not be affected.
+
+## {% linkable_title Security %}
```
Comment (balloob) [inline source/_ecosystem/ios/notifications/privacy_security_rate_limits.markdown]: Header without section?

File: source/_ecosystem/ios/notify.markdown [hunk 9]
```diff
@@ -0,0 +1,423 @@
+---
```
Comment (balloob) [inline source/_ecosystem/ios/notify.markdown]: This page is not referenced?

File: source/_ecosystem/ios/requirements.markdown [hunk 10]
```diff
@@ -0,0 +1,14 @@
+---
+layout: page
+title: "Requirements"
```
Comment (balloob) [inline source/_ecosystem/ios/requirements.markdown]: This seems like a page that can be part of some other page, maybe getting started ?

File: source/_ecosystem/ios/security.markdown [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+title: "Security"
+description: "Describes Home Assistant for iOS security"
+date: 2016-10-25 15:00:00 -0700
+sidebar: true
+comments: false
+sharing: true
+footer: true
+---
+
+* Please use SSL.
```
Comment (balloob) [inline source/_ecosystem/ios/security.markdown]: This page can get integrated too with some other page ?

File: source/_ecosystem/ios/support.markdown [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+title: "Support"
+description: "Home Assistant for iOS support"
+date: 2016-10-25 15:00:00 -0700
+sidebar: true
+comments: false
+sharing: true
+footer: true
+---
+
+[Forums](https://community.home-assistant.io/c/ios)
```
Comment (balloob) [inline source/_ecosystem/ios/support.markdown]: Not linked

File: source/_ecosystem/ios/sensor.markdown [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+title: "Sensors"
+description: "Home Assistant for iOS battery sensors"
+date: 2016-10-25 15:00:00 -0700
+sidebar: true
+comments: false
+sharing: true
+footer: true
+---
+
+As of Home Assistant 0.31.1 you will find battery state and level sensors for each registered device.
```
Comment (balloob) [inline source/_ecosystem/ios/sensor.markdown]: Does not warrant a full page ?

File: source/_ecosystem/ios/notifications/content_extensions.markdown (hunk 7 above)
Comment (robbiet480) [inline source/_ecosystem/ios/notifications/content_extensions.markdown]: I don't understand this comment

**Comment by robbiet480** [comment]
@balloob Still don't understand [this](https://github.com/home-assistant/home-assistant.github.io/pull/1312#discussion_r85469736) comment

File: source/_ecosystem/ios/notifications/content_extensions.markdown (hunk 7 above)
Comment (balloob) [inline source/_ecosystem/ios/notifications/content_extensions.markdown]: Why don't we merge this page together with attachments and just make the page "Advanced Notifications" ?

**Pull request merged**